lib/logstash/inputs/http_poller.rb

```diff
@@ -164,7 +190,7 @@ def handle_success(queue, name, request, response, execution_time)
     # responses come up as "" which will cause the codec to not yield anything
     if body && body.size > 0
       decode_and_flush(@codec, body) do |decoded|
-        event = @target ? LogStash::Event.new(@target => decoded.to_hash) : decoded
+        event = @target ? targeted_event_factory.new_event(decoded.to_hash) : decoded
```

Comment:
:heart: could we leverage the event_factory in other places as well, please?
e.g. changing [LogStash::Event.new](https://github.com/logstash-plugins/logstash-input-http_poller/blob/v5.0.2/lib/logstash/inputs/http_poller.rb#L171) to `event_factory.new_event`